wgpu/tests/partially_bounded_arrays/mod.rs

```diff
@@ -0,0 +1,156 @@
+                label: Some("main"),
+                bind_group_layouts: &[&bind_group_layout],
+                push_constant_ranges: &[],
+            });
+
+            let compute_pipeline =
+                device.create_compute_pipeline(&wgpu::ComputePipelineDescriptor {
+                    label: None,
+                    layout: Some(&pipeline_layout),
+                    module: &cs_module,
+                    entry_point: "main",
+                });
+
+            let bind_group = device.create_bind_group(&wgpu::BindGroupDescriptor {
+                entries: &[wgpu::BindGroupEntry {
+                    binding: 0,
+                    resource: wgpu::BindingResource::TextureViewArray(&[&texture_view]),
+                }],
+                layout: &bind_group_layout,
+                label: Some("bind group"),
+            });
+
+            let mut encoder =
+                device.create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+            {
+                let mut cpass =
+                    encoder.begin_compute_pass(&wgpu::ComputePassDescriptor { label: None });
+                cpass.set_pipeline(&compute_pipeline);
+                cpass.set_bind_group(0, &bind_group, &[]);
+                cpass.dispatch_workgroups(1, 1, 1);
+            }
+
+            encoder.copy_texture_to_buffer(
+                wgpu::ImageCopyTexture {
+                    texture: &storage_texture,
+                    mip_level: 0,
+                    origin: wgpu::Origin3d::ZERO,
+                    aspect: wgpu::TextureAspect::All,
+                },
+                wgpu::ImageCopyBuffer {
+                    buffer: &staging_buffer,
+                    layout: wgpu::ImageDataLayout {
+                        offset: 0,
+                        bytes_per_row: None,
+                        rows_per_image: None,
+                    },
+                },
+                wgpu::Extent3d {
+                    width: 1,
+                    height: 1,
+                    depth_or_array_layers: 1,
+                },
+            );
+
+            ctx.queue.submit(Some(encoder.finish()));
+
+            // wait for gpu
+            let buffer_slice = staging_buffer.slice(..);
+            let (sender, receiver) = futures_intrusive::channel::shared::oneshot_channel();
+            buffer_slice.map_async(wgpu::MapMode::Read, move |v| sender.send(v).unwrap());
+
+            device.poll(wgpu::Maintain::Wait);
+
+            if let Some(Ok(())) = pollster::block_on(receiver.receive()) {
+                let data = buffer_slice.get_mapped_range();
+                let result: Vec<f32> = bytemuck::cast_slice(&data).to_vec();
+                assert!(result.iter().eq(&[4.0, 3.0, 2.0, 1.0]));
+                // dropped before we unmap the buffer.
+                drop(data);
+                staging_buffer.unmap();
+            } else {
+                panic!("failed!")
+            }
```

Comment:
I think most of this should be able to be replaced with the ReadbackBuffers here: https://github.com/gfx-rs/wgpu/blob/trunk/wgpu/tests/common/image.rs#L350-L355